### PR TITLE
Refactored property relationships

### PR DESCRIPTION
**Pull request by stuartio**
Many fixes

### CODE REVIEW

File: nodestream_akamai/akamai_utils/cloudlets_v2_client.py [hunk 1]
```diff
@@ -78,7 +95,7 @@ def extract_active_akamai_redirect_policy_versions(self, policy_tree):
         if len(list_of_activations) == 0:
             logger.info(
                 "Policy had no active occurrences. Candidate for cleanup: Policy ID %s",
-                policy_tree["policyId"],
+                str(policy_tree["policyId"]),
```
Comment (zprobst) [inline nodestream_akamai/akamai_utils/cloudlets_v2_client.py:98]: The `%s` formatter should format the number without needing to call `str`. 

```suggestion
                policy_tree["policyId"],
```

File: nodestream_akamai/akamai_utils/cloudlets_v2_client.py [hunk 2]
```diff
@@ -93,7 +110,7 @@ def get_policy_rule_set(self, policy):
                     raw_ruleset = self.extract_akamai_ruleset_version(
                         policy["policyId"], policy["version"]
                     )
-                    policy["id"] = f"akamai_redirect:{policy['policyId']}"
+                    policy["id"] = f"akamai_redirect:{str(policy['policyId'])}"
```
Comment (zprobst) [inline nodestream_akamai/akamai_utils/cloudlets_v2_client.py:113]: Same here 

```suggestion
                    policy["id"] = f"akamai_redirect:{policy['policyId']}"
```

File: nodestream_akamai/akamai_utils/cloudlets_v2_client.py [hunk 3]
```diff
@@ -102,10 +119,9 @@ def get_policy_rule_set(self, policy):
                     ] = self.search_akamai_ruleset_for_outbound_hosts(raw_ruleset)
                     policy_list.append(policy)
         except Exception as e:
-            logger.info("No version found in: %s", policy["policyId"])
+            logger.info("No version found in: %s", str(policy["policyId"]))
```
Comment (zprobst) [inline nodestream_akamai/akamai_utils/cloudlets_v2_client.py:122]: Same here

```suggestion
            logger.info("No version found in: %s", policy["policyId"])
```

File: nodestream_akamai/akamai_utils/model.py [hunk 4]
```diff
@@ -30,24 +34,24 @@ def origin_count(self):
     def hostname_count(self):
         return len(self.hostnames)
 
-    @property
-    def cloudlet_policy_count(self):
-        return {"edgeRedirector": len(self.cloudlet_policies["edgeRedirector"])}
+    # @property
+    # def cloudlet_policy_count(self):
+    #     return {"edgeRedirector": len(self.cloudlet_policies["edgeRedirector"])}
```
Comment (zprobst) [inline nodestream_akamai/akamai_utils/model.py:39]: Shall we remove this?

File: nodestream_akamai/siteshield.yaml [hunk 5]
```diff
@@ -1,27 +1,26 @@
...
-      node_type: AkamaiSiteshieldConfig
+      node_type: AkamaiSiteshieldMap
       normalization:
         do_remove_trailing_dots: true
       key:
-        id: !jmespath 'id'
+        rule_name: !jmespath 'ruleName'
     - type: properties
       properties:
+        id: !jmespath 'id'
```
Comment (zprobst) [inline nodestream_akamai/siteshield.yaml:19]: Why do we want to derive the key from `rule_name` as opposed to `id`. Intuitively, seems like `id` is a better choice?

File: nodestream_akamai/netstorage-group.yaml [hunk 6]
```diff
@@ -24,50 +21,56 @@
         domain_prefix: !jmespath 'domainPrefix'
     - type: relationship
       node_type: Endpoint
-      relationship_type: HAS_RECORD
+      relationship_type: UPLOADS_TO
```
Comment (zprobst) [inline nodestream_akamai/netstorage-group.yaml:24]: We did notice this oddity the other day. Nice change 👍

File: nodestream_akamai/gtm/gtm.py [hunk 7]
```diff
@@ -2,28 +2,44 @@
...
                     "name": raw_domain["name"],
-                    "summary": summary,
-                    "raw": raw_domain,
+                    "properties": properties,
+                    "type": raw_domain["type"],
+                    "loadImbalancePercentage": raw_domain["loadImbalancePercentage"],
+                    "loadFeedback": raw_domain["loadFeedback"],
+                    "cnameCoalescingEnabled": raw_domain["cnameCoalescingEnabled"],
+                    "deeplink": deeplink
+                    # "raw": raw_domain,
```
Comment (zprobst) [inline nodestream_akamai/gtm/gtm.py:42]: Can this be removed?

**Review by zprobst — COMMENTED** [review]
Some minor comments - but looking good.  

Let me know what you think so I can take one last pass and get this merged 👍

File: nodestream_akamai/gtm/gtm.py (hunk 7 above)
Comment (stuartio) [reply nodestream_akamai/gtm/gtm.py:42]: Why? These were just additional properties I felt might be useful in the GTM_DOMAIN object

File: nodestream_akamai/akamai_utils/model.py (hunk 4 above)
Comment (stuartio) [reply nodestream_akamai/akamai_utils/model.py:39]: I just left it in for legacy reasons. The now split between regular cloudlets and redirect configs is something I need to discuss with you guys

File: nodestream_akamai/siteshield.yaml (hunk 5 above)
Comment (stuartio) [reply nodestream_akamai/siteshield.yaml:19]: So this was a weird one. When the key of the node was id, the relationships didnt work between Siteshield maps and properties. But, when I changed it to the rule_name, which is also unique so could work as a key, the relationships were fine. I don't know why, it doesn't make sense to me, but I suspect the issue was the Siteshield nodes were created by the relationship with the property _first_, and thus had their key as the rule_name. Then when the Siteshield pipeline ran it created new nodes with the id as they key, and I ended up with duplicate nodes. Does that track?

File: nodestream_akamai/gtm/gtm.py (hunk 7 above)
Comment (zprobst) [reply nodestream_akamai/gtm/gtm.py:42]: I just meant the comment. The properties are all good 👍

Comment (stuartio) [reply nodestream_akamai/gtm/gtm.py:42]: Ah I see. Yes, lemme pull that

**Review by zprobst — APPROVED** [review]
👍 

This is a significant evolution

**Pull request merged**